src/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -1179,6 +1181,15 @@ protected SAMBinaryTagAndValue getBinaryAttributes() {
         return mAttributes;
     }
 
+    @Override
+    public SimpleInterval getInterval() {
+        if (mReferenceName != null && mAlignmentStart > 0){
```

Comment:
What happens to unmapped reads with a reference and start but no alignment end?  I.e. they are given the start coordinate of their mapped mate.